ingestion/src/metadata/ingestion/source/database/datalake.py

```diff
@@ -0,0 +1,315 @@
+            )
+        return cls(config, metadata_config)
+
+    def prepare(self):
+        pass
+
+    def next_record(self) -> Iterable[Entity]:
+        try:
+
+            bucket_name = self.service_connection.base_path
+            print(bucket_name)
+            yield from self.get_gcs_files(bucket_name)
+            # yield from self.get_s3_files(bucket_name)
+
+        except Exception as err:
+            logger.error(traceback.format_exc())
+            logger.error(err)
+
+    # def get_hdfs_files(self, bucket_name):
+    #     #csv files
+    #     sparkSession = SparkSession.builder.appName("example-pyspark-read-and-write").getOrCreate()
+    #     df = sparkSession.read.csv('hdfs://cluster/user/hdfs/test/example.csv')
+
+    #     #json files
+    #     df = sparkSession.read.json("resources/zipcodes.json")
+
+    #     #tsv files
+    #     df = sparkSession.read.csv("path", sep=r'\t', header=True)
+
+    def get_gcs_files(self, bucket_name):
+        storage_client = storage.Client()
+        bucket = storage_client.get_bucket(bucket_name)
+
+        for key in bucket.list_blobs():
+            print(key.name)
+            if key.name.endswith("csv"):
+                df = dd.read_csv(f"gs://{bucket_name}/{key.name}")
+                print(df)
+                if filter_by_table(
+                    self.config.sourceConfig.config.tableFilterPattern, key.name
+                ):
+                    self.status.filter(
+                        "{}".format(key["Key"]),
+                        "Table pattern not allowed",
+                    )
+                    continue
+                yield from self.ingest_tables(key.name, df, bucket_name)
+
+            if key.name.endswith("tsv"):
+                df = dd.read_csv(f"gs://{bucket_name}/{key.name}")
+
+                if filter_by_table(
+                    self.config.sourceConfig.config.tableFilterPattern, key.name
+                ):
+                    self.status.filter(
+                        "{}".format(key["Key"]),
+                        "Table pattern not allowed",
+                    )
+                    continue
+                yield from self.ingest_tables(key.name, df, bucket_name)
+
+            if key.name.endswith("json"):
+                data = key.download_as_string().decode()
+                df = pd.DataFrame.from_dict(data)
+                if filter_by_table(
+                    self.config.sourceConfig.config.tableFilterPattern, key.name
+                ):
+                    self.status.filter(
+                        "{}".format(key["Key"]),
+                        "Table pattern not allowed",
+                    )
+                    continue
+                yield from self.ingest_tables(key.name, df, bucket_name)
+
+    def get_s3_files(self, bucket_name):
+        for key in self.s3.list_objects(Bucket=bucket_name)["Contents"]:
+            try:
+                if key["Key"].endswith("csv"):
```

Comment:
maybe it is never needed, but might be safer to check `endswith(".csv")`, adding a dot. We can have this for all formats